internal/jwtparser/jwt_test.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jwtparser_test
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"net/http"
+	"net/http/httptest"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/juju/errors"
+	jc "github.com/juju/testing/checkers"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/internal/jwtparser"
+)
+
+type jwtParserSuite struct {
+	url        string
+	keySet     jwk.Set
+	signingKey jwk.Key
+	srv        *httptest.Server
```

Comment:
We don't need this. The dep is only a http client that has a `Get` method. We can mock this out and returned a http response that way. We don't need to introduce this flakiness into our tests and all the problems it can occur.